src/core/EdFi.DataManagementService.Core/Response/FailureResponse.cs

```diff
@@ -92,4 +93,29 @@ public static BaseFailureResponse ForInvalidReferences(ResourceName[] resourceNa
             correlationId: null
         );
     }
+
+    public static FailureResponseWithErrors ForImmutableIdentity(ResourceName[] resourceNames) =>
```

Comment:
This is not the correct behavior of ForImmutableIdentity, ForKeyChangeNotSupported is. This is identical to ForInvalidReferences behavior directly above. Delete this unnecessary method.